python/lsst/daf/butler/registry/queries/expressions/parser/exprTree.py

```diff
@@ -73,6 +73,14 @@
 
     from .treeVisitor import TreeVisitor
 
+
+def _strip_parens(expression: Node) -> Node:
```

Comment:
I'm missing something here because it looks like this function only calls itself and does nothing to actually strip any parens...

Reply:
It should strip parens. If the node is not `Parens` it returns the node itself, but if the node is `Parens` then it returns the child of the node (recursively) effectively removing parens. This works on a parsed expression tree, this is why you do not see any `()` here.